state/executor.go

```diff
@@ -574,7 +578,7 @@ func (t *Transition) apply(msg *types.Transaction) (*runtime.ExecutionResult, er
 	// We use EIP-1559 fields of the tx if the london hardfork is enabled.
 	// Effective tip be came to be either gas tip cap or (gas fee cap - current base fee)
 	effectiveTip := new(big.Int).Set(gasPrice)
-	if t.config.London && msg.Type != types.StateTx {
+	if t.config.London && msg.Type == types.DynamicFeeTx {
```

Comment:
There is a condition in this `apply` function:

```
// Burn some amount if the london hardfork is applied.
// Basically, burn amount is just transferred to the current burn contract.
if t.config.London && msg.Type != types.StateTx {
```

Should we change it to check if `msg.Type == types.DynamicFeeTx` like we did here? Maybe we can move it in this if block.

Reply:
The idea here is that there are not gas fee cap and gas tip cap in non-dynamic-gas-fee tx so it jus does not make any sense to apply this check for legacy or state txs.